src/context/auth-context.js

```diff
@@ -0,0 +1,56 @@
+import { createContext, useContext, useState } from 'react';
+import { login, signup } from '../util/auth-request';
+
+const AuthContext = createContext();
+
+const currentUser = 'ANI_SHOPPIE_USER';
+
+const AuthProvider = ({ children }) => {
+	const initialUser = JSON.parse(localStorage.getItem(currentUser)) || {};
+	const [user, setUser] = useState(initialUser);
+
+	const handleLogin = async ({ email, password }) => {
+		try {
+			const {
+				data: { foundUser: user, encodedToken: token }
+			} = await login({ email, password });
+			if (token) {
+				localStorage.setItem(currentUser, JSON.stringify({ user, token }));
+				setUser(JSON.parse(localStorage.getItem(currentUser)));
+			}
+		} catch (err) {
+			console.error(err);
+		}
+	};
+
+	const handleSignUp = async ({ name, email, password }) => {
+		try {
+			const {
+				data: { createdUser: user, encodedToken: token }
+			} = await signup({ name, email, password });
+			if (token) {
+				localStorage.setItem(currentUser, JSON.stringify({ user, token }));
+				setUser(JSON.parse(localStorage.getItem(currentUser)));
+			}
+		} catch (err) {
+			console.error(err);
+		}
+	};
+
+	const handleLogout = () => {
+		localStorage.removeItem(currentUser);
+		setUser({});
+	};
+
+	return (
+		<AuthContext.Provider
+			value={{ user, handleLogin, handleLogout, handleSignUp }}
```

Comment:
Just a suggestion, you can create an object and put the properties there, and then pass the obj to value. This way you will pass only one value instead of passing no. of values and also it will look much cleaner.

Reply:
Thanks for the suggestion will work on it.